### PR TITLE
fix: should now use the activated environment when running a task

### PR DESCRIPTION
**Pull request by tdejager**
Use the recently-added logic for figuring out the env name from an activated shell. Should fix: https://github.com/prefix-dev/pixi/issues/1435

### CODE REVIEW

**Review by baszalmstra — APPROVED** [review]

File: src/cli/run.rs [hunk 1]
```diff
@@ -63,16 +62,18 @@ pub async fn execute(args: Args) -> miette::Result<()> {
     verify_prefix_location_unchanged(project.default_environment().dir().as_path()).await?;
 
     // Extract the passed in environment name.
-    let explicit_environment = args
-        .environment
-        .map(|n| EnvironmentName::from_str(n.as_str()))
-        .transpose()?
-        .map(|n| {
+    let env_name = EnvironmentName::from_arg_or_env_var(args.environment).into_diagnostic()?;
```
Comment (ruben-arts) [inline src/cli/run.rs:65]: isn't it possible to do this:
```rs
let explicit_environment = project.environment_from_name_or_env_var(args.environment)?;
```

Comment (ruben-arts) [reply src/cli/run.rs:65]: Now all tools that use this function also warn the user at the end that it was used:
```rs
Project::warn_on_discovered_from_env(args.manifest_path.as_deref());
```

Comment (ruben-arts) [reply src/cli/run.rs]: Ah sorry didn't check as I assumed it was never added. This feels like a merge conflict fix.

**Review by ruben-arts — APPROVED** [review]
Thanks Tim!

**Pull request merged**